src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyMetaPackages.cs

```diff
@@ -30,15 +31,33 @@ public class ApplyMetaPackages : PackagingTask
         /// Package index files used to meta-package mappings
         /// </summary>
         public ITaskItem[] PackageIndexes { get; set; }
+
+        /// <summary>
+        /// List of package IDs which should be suppressed from remapping
+        /// </summary>
+        public string[] SuppressMetaPackages { get; set; }
+
+        /// <summary>
+        /// Set to true to apply the meta-package remapping
+        /// </summary>
+        public bool Apply { get; set; }
         
         [Output]
         public ITaskItem[] UpdatedDependencies { get; set; }
         
         public override bool Execute()
         {
+            if (!Apply)
+            {
+                UpdatedDependencies = OriginalDependencies;
+                return true;
+            }
+
             var index = PackageIndex.Load(PackageIndexes.Select(pi => pi.GetMetadata("FullPath")));
             List<ITaskItem> updatedDependencies = new List<ITaskItem>();
 
+            var suppressMetaPackages = new HashSet<string>(SuppressMetaPackages.NullAsEmpty(), StringComparer.OrdinalIgnoreCase);
+
             // We cannot add a dependency to a meta-package from a package that itself is part of the meta-package otherwise we create a cycle
             var metaPackageThisPackageIsIn = index.MetaPackages.GetMetaPackageId(PackageId);
```

Comment:
You could use the same mechanism for this case and just add metaPackageThisPackageIsIn to your suppressMetaPackages set.